src/coreclr/jit/gcinfo.cpp

```diff
@@ -252,6 +252,12 @@ GCInfo::WriteBarrierForm GCInfo::gcIsWriteBarrierCandidate(GenTreeStoreInd* stor
         return WBF_NoBarrier;
     }
 
+    if (store->Data()->IsIconHandle(GTF_ICON_OBJ_HDL))
```

Comment:
I am not an expert in `GTF_` things, but I assume this condition will not hold if an object was allocated on POH. 

For POH objects we need to run the barrier.

Reply:
For now, this flag is only used for these:
1) string literals (only in non-collectible contexts)
2) RuntimeType (only in non-collectible contexts)

and only for FOH, we don't special case POH